pkg/apis/admissionregistration/types.go

```diff
@@ -334,6 +339,18 @@ type Validation struct {
 	MessageExpression string
 }
 
+// Variable is the definition of a variable that is used for composition.
+// +structType=atomic
+type Variable struct {
+	// Name is the name of the variable. The variable can be accessed in other expressions through `variables`
+	// For example, if name is "foo", the variable will be available as `variables.foo`
+	Name string
+
+	// Expression is the expression that will be evaluated as the value of the variable.
+	// The syntax corresponds with that of a Validation.
```

Comment:
```suggestion
	// The CEL expression has access to the same variables as the validation CEL expressions.
```

Reply:
changing `variables` to `identifiers` to avoid the term `variables` being overloaded.